src/modules/map/components/cesium/CesiumWMTSLayer.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+    <div>
+        <slot />
+    </div>
+</template>
+
+<script>
+import GeoAdminWMTSLayer from '@/api/layers/GeoAdminWMTSLayer.class'
+import { CoordinateSystem, WEBMERCATOR } from '@/utils/coordinateSystems'
+import { ImageryLayer, Rectangle, UrlTemplateImageryProvider } from 'cesium'
+import { WGS84_EXTENT } from '@/config'
+import addImageryLayerMixins from './utils/addImageryLayer-mixins'
+
+// todo should we adapt LOD to resolution?
+const MAXIMUM_LEVEL_OF_DETAILS = 18
+
+export default {
+    mixins: [addImageryLayerMixins],
+    props: {
+        wmtsLayerConfig: {
+            type: GeoAdminWMTSLayer,
+            required: true,
+        },
+        previewYear: {
+            type: Number,
+            default: null,
+        },
+        projection: {
+            type: CoordinateSystem,
+            default: WEBMERCATOR,
+        },
+        zIndex: {
+            type: Number,
+            default: -1,
+        },
+    },
+    computed: {
+        layerId() {
+            return this.wmtsLayerConfig.getID()
+        },
+        opacity() {
+            return this.wmtsLayerConfig.opacity || 1.0
+        },
+        timestampForPreviewYear() {
+            if (
+                this.previewYear &&
+                this.wmtsLayerConfig.timeConfig &&
+                this.wmtsLayerConfig.timeConfig.years.includes(this.previewYear)
+            ) {
+                return this.wmtsLayerConfig.timeConfig.getTimeEntryForYear(this.previewYear)
+                    .timestamp
+            }
+            return null
+        },
```

Comment:
Maybe we should move this logic too, as it is the same as the one found in `src/modules/map/components/openlayers/OpenLayersWMTSLayer.vue`

I don't know if this is about the same as what is done in WMS, but if it is the case it could be a good opportunity to condense these two into one single function